noir-projects/Earthfile

```diff
@@ -102,13 +107,16 @@ format:
   RUN nargo fmt --check
 
 gates-report:
-  FROM +build-protocol-circuits
+  FROM +source
   WORKDIR /usr/src/noir-projects
 
-  COPY ./gates_report.sh ./gates_report.sh
-  COPY ../barretenberg/cpp/+preset-clang-assert/bin/bb /usr/src/barretenberg/cpp/build/bin/bb
-
+  # Install bb
+  COPY ../barretenberg/cpp/+preset-release/bin/bb /usr/src/barretenberg/cpp/build/bin/bb
   ENV BB_BIN /usr/src/barretenberg/cpp/build/bin/bb
+
+  RUN cd noir-protocol-circuits && yarn && node ./generate_variants.js && nargo compile --silence-warnings
```

Comment:
nit: could we have a separate job that just compiles the protocol circuits without key gen. Then that job becomes the target of the `gates-report` and `build-protocol-circuits`

Reply:
Yep but that gets into needing to restructure the bootstrap scripts so it's not doing everything. We're way more likely to get backsliding in future as other people change things if we go down that route (as well as it being more invasive) so this seemed cleaner.